### PR TITLE
Add installation setup for istio and Eventing

### PR DESCRIPTION
**Pull request by pierDipi**
You can run:
```
hack/install-dependencies.sh
```
to get Istio and Eventing with istio sidecar injected

In KinD I get
```
$ k get pods -A
NAMESPACE            NAME                                          READY   STATUS    RESTARTS        AGE
istio-system         grafana-56bdf8bf85-4p2fq                      1/1     Running   0               7m12s
istio-system         istio-ingressgateway-6785fcd48-tbsjl          1/1     Running   0               7m58s
istio-system         istiod-65448977c9-gg8fv                       1/1     Running   0               8m50s
istio-system         jaeger-c4fdf6674-j4xks                        1/1     Running   0               7m12s
istio-system         kiali-849958788-ndg2w                         1/1     Running   0               7m12s
istio-system         prometheus-85949fddb-nkfjx                    2/2     Running   0               7m12s
kafka                my-cluster-entity-operator-5d955b4cb5-65785   2/3     Running   0               78s
kafka                my-cluster-kafka-0                            1/1     Running   0               104s
kafka                my-cluster-kafka-1                            1/1     Running   0               104s
kafka                my-cluster-kafka-2                            1/1     Running   0               104s
kafka                my-cluster-zookeeper-0                        1/1     Running   2 (2m27s ago)   4m59s
kafka                my-cluster-zookeeper-1                        1/1     Running   0               4m59s
kafka                my-cluster-zookeeper-2                        1/1     Running   0               4m59s
kafka                strimzi-cluster-operator-694dddcf8c-cldp2     1/1     Running   0               7m10s
knative-eventing     eventing-controller-6fdd84c577-n6lwc          1/1     Running   0               8m53s
knative-eventing     eventing-webhook-55968f89f5-pxjqp             1/1     Running   0               8m52s
knative-eventing     imc-controller-595d7647df-dsd75               1/1     Running   0               8m52s
knative-eventing     imc-dispatcher-688db6945-m2q47                1/1     Running   0               8m53s
knative-eventing     kafka-broker-dispatcher-85957bc9b7-b64nr      2/2     Running   0               7m12s
knative-eventing     kafka-broker-receiver-8d8fd479d-zpzgs         2/2     Running   0               7m12s
knative-eventing     kafka-channel-dispatcher-64bc55bff8-7qcvv     2/2     Running   0               7m12s
knative-eventing     kafka-channel-receiver-5c6847547c-g9vcm       2/2     Running   0               7m12s
knative-eventing     kafka-controller-69bf748484-kjmc8             2/2     Running   0               7m11s
knative-eventing     kafka-sink-receiver-688f89ff4d-8x5r5          2/2     Running   0               7m11s
knative-eventing     kafka-webhook-eventing-775cfb5cf5-ndbpv       1/1     Running   0               7m11s
knative-eventing     mt-broker-controller-86b7485cff-4wfd9         1/1     Running   0               8m52s
knative-eventing     mt-broker-filter-857f876946-s8lwx             1/1     Running   0               8m52s
knative-eventing     mt-broker-ingress-6ccf8b8947-pf4rm            1/1     Running   0               8m52s
kube-system          coredns-565d847f94-742vw                      1/1     Running   0               8m53s
kube-system          coredns-565d847f94-9frhd                      1/1     Running   0               8m53s
kube-system          etcd-kind-control-plane                       1/1     Running   0               9m8s
kube-system          kindnet-l2h74                                 1/1     Running   0               8m53s
kube-system          kube-apiserver-kind-control-plane             1/1     Running   0               9m7s
kube-system          kube-controller-manager-kind-control-plane    1/1     Running   0               9m7s
kube-system          kube-proxy-4mv8d                              1/1     Running   0               8m53s
kube-system          kube-scheduler-kind-control-plane             1/1     Running   0               9m6s
local-path-storage   local-path-provisioner-684f458cdd-9ndnf       1/1     Running   0               8m53s

```

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

File: hack/install-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  --patch-file "${REPO_ROOT_DIR}/hack/eventing-injection-disabled.yaml"
+
+kubectl patch deployment \
+  imc-controller \
+  --type merge \
+  -n knative-eventing \
+  --patch-file "${REPO_ROOT_DIR}/hack/eventing-injection-disabled.yaml"
+
+echo "Installing Istio"
+istioctl version
```
Comment (matzew) [inline hack/install-dependencies.sh:33]: any specific version we want to call out as a prequisit to this?

**Comment by matzew** [comment]
How do you start `kind`? 
I had to tweak `/etc/sysctl.conf` file (`too many open files`), on my new machine, but still had to set Kafka to a single node cluster :smile: 

But this script works, and install all the things

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by pierDipi** [comment]
`kind create cluster`

File: hack/install-dependencies.sh (hunk 1 above)
Comment (pierDipi) [reply hack/install-dependencies.sh:33]: No, to install the version we will use you can run `hack/update-istio.sh

**Pull request merged**